spyder_kernels/console/kernel.py

```diff
@@ -13,9 +13,11 @@
 # Standard library imports
 import os
 import sys
+import threading
 
 # Third-party imports
 from ipykernel.ipkernel import IPythonKernel
+from ipython_genutils.py3compat import PY3, input
```

Comment:
`ipykernel` could drop this dependency once it moves to support Python 3 only. So we either declare this dependency explicitly in our `setup.py` or we add the `input` object to our own `py3compat` (I would prefer that).